test/integration/apiserver/apply/apply_test.go

```diff
@@ -145,6 +145,95 @@ func TestApplyAlsoCreates(t *testing.T) {
 	}
 }
 
+// TestNoOpUpdateSameResourceVersion makes sure that PUT requests which change nothing
+// will not change the resource version (no write to etcd is done)
+func TestNoOpUpdateSameResourceVersion(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	podName := "no-op"
+	podResource := "pods"
+	podBytes := []byte(`{
+		"apiVersion": "v1",
+		"kind": "Pod",
+		"metadata": {
+			"name": "` + podName + `"
+		},
+		"spec": {
```

Comment:
Would it make sense to have a more sophisticated object so that we know that maps, associative list and everything keep things in order?

Reply:
We can do that, but as it is, the ownership already looks like:
```
"fields": {
	"f:spec": {
		"f:containers": {
			"k:{\"name\":\"test-container\"}": {
				"f:image": {},
				"f:imagePullPolicy": {},
				"f:name": {},
				"f:resources": {},
				"f:terminationMessagePath": {},
				"f:terminationMessagePolicy": {},
				".": {}
			}
		},
		"f:dnsPolicy": {},
		"f:enableServiceLinks": {},
		"f:restartPolicy": {},
		"f:schedulerName": {},
		"f:securityContext": {},
		"f:terminationGracePeriodSeconds": {}
	}
}
```

Reply:
Great, multiple containers does help

Reply:
multiple containers and a few labels should cover most of the interesting features